core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -47,55 +51,48 @@
   private int numRecords;
   private int numRecordsRemaining;
 
-  private byte[] arr = new byte[1024 * 1024];
+  private byte[] arr = new byte[1024 * bufferMultiplicationRatio];
```

Comment:
To reflect the runtime config (`spark.unsafe.sorter.spill.reader.buffer.size.ratio`), how about moving this array init into the constructor?